python/pyarrow/tests/parquet/test_metadata.py

```diff
@@ -533,6 +536,44 @@ def test_metadata_exceeds_message_size():
     metadata = pq.read_metadata(pa.BufferReader(buf))
 
 
+def test_metadata_schema_filesystem(tmpdir):
+    table = pa.table({"a": [1, 2, 3]})
+
+    # URI writing to local file.
+    fname = "data.parquet"
+    file_path = 'file:///' + os.path.join(str(tmpdir), fname)
+
+    pq.write_table(table, file_path)
+
+    # Get expected `metadata` from path.
+    metadata = pq.read_metadata(tmpdir / fname)
+    schema = table.schema
+
+    assert pq.read_metadata(file_path).equals(metadata)
+    assert pq.read_metadata(
+        fname, filesystem=f'file:///{tmpdir}').equals(metadata)
+
+    assert pq.read_schema(file_path).equals(schema)
+    assert pq.read_schema(fname, filesystem=f'file:///{tmpdir}').equals(schema)
+
+    with util.change_cwd(tmpdir):
+        # Pass `filesystem` arg
+        assert pq.read_metadata(
+            fname, filesystem=LocalFileSystem()).equals(metadata)
+        assert pq.read_metadata(
+            fname, filesystem=LocalFileSystem.get_instance()).equals(metadata)
```

Comment:
The new LocalFileSystem has no `get_instance()` method, so I think you can just remove this case (since the line above already includes the case of a plain LocalFileSystem())

Reply:
Thanks! I was away from keyboard for the week. Will fix it!